--- leafnode+-2.14/Makefile	Wed Jun 13 19:53:27 2001
+++ leafnode+-2.14.new/Makefile	Wed Sep 19 17:30:10 2001
@@ -9,7 +9,7 @@
 #CC = gcc
 #CC = cc
 #AR = ar
-#RANLIB = ranlib
+RANLIB = ranlib
 # Programs for distribution.
 CP = cp
 CHMOD = chmod
@@ -24,6 +24,7 @@
 PREFIX = /usr/local
 BINDIR = $(PREFIX)/sbin
 MANDIR = $(PREFIX)/man
+DESTDIR =
 #
 # If you want to build with debugging, use below.
 #CFLAGS += -g -Wall -Wformat -Wstrict-prototypes -Wmissing-prototypes 
@@ -112,41 +113,29 @@
 		-DVERSION=\"$(VERSION)\" $<
 
 install: leafnode fetchnews texpire checkgroups leaftool
-	$(CP) leafnode $(BINDIR)/leafnode
-	$(CP) fetchnews $(BINDIR)/fetchnews
-	$(CP) texpire $(BINDIR)/texpire
-	$(CP) checkgroups $(BINDIR)/checkgroups
-	$(CP) leaftool $(BINDIR)/leaftool
-	$(CHMOD) 750 $(BINDIR)/leafnode
-	$(CHMOD) 750 $(BINDIR)/fetchnews
-	$(CHMOD) 750 $(BINDIR)/texpire
-	$(CHMOD) 750 $(BINDIR)/checkgroups
-	$(CHMOD) 750 $(BINDIR)/leaftool
-	$(CHOWN) news.news $(BINDIR)/leafnode
-	$(CHOWN) news.news $(BINDIR)/fetchnews
-	$(CHOWN) news.news $(BINDIR)/texpire
-	$(CHOWN) news.news $(BINDIR)/checkgroups
-	$(CHOWN) news.news $(BINDIR)/leaftool
-	$(CP) *.8 $(MANDIR)/man8
+	$(CP) leafnode $(DESTDIR)$(BINDIR)/leafnode
+	$(CP) fetchnews $(DESTDIR)$(BINDIR)/fetchnews
+	$(CP) texpire $(DESTDIR)$(BINDIR)/texpire
+	$(CP) checkgroups $(DESTDIR)$(BINDIR)/checkgroups
+	$(CP) leaftool $(DESTDIR)$(BINDIR)/leaftool
+	$(CP) *.8 $(DESTDIR)$(MANDIR)/man8
 	@if [ ! -d $(SPOOLDIR)/message.id ]; then echo don"'"t forget to do '"'make installall'"' to create $(SPOOLDIR)/message.id; fi
 	@if [ -f $(BINDIR)/fetch ]; then echo don"'"t forget to change the name of a command in scripts and crontab from fetch into fetchnews.; fi
 	@if [ -f $(BINDIR)/newsfetch ]; then echo don"'"t forget to change the name of a command in scripts and crontab from newsfetch into fetchnews.; fi
 
 installall: install
-	-$(MKDIR) $(SPOOLDIR) $(LIBDIR)
-	-$(MKDIR) $(SPOOLDIR)/message.id
-	-$(MKDIR) $(SPOOLDIR)/interesting.groups
-	-$(MKDIR) $(SPOOLDIR)/out.going
-	cd $(SPOOLDIR)/message.id ; for a in 0 1 2 3 4 5 6 7 8 9 ; do \
+	-$(MKDIR) $(DESTDIR)$(SPOOLDIR) $(DESTDIR)$(LIBDIR)
+	-$(MKDIR) $(DESTDIR)$(SPOOLDIR)/message.id
+	-$(MKDIR) $(DESTDIR)$(SPOOLDIR)/interesting.groups
+	-$(MKDIR) $(DESTDIR)$(SPOOLDIR)/out.going
+	cd $(DESTDIR)$(SPOOLDIR)/message.id ; for a in 0 1 2 3 4 5 6 7 8 9 ; do \
 		for b in 0 1 2 3 4 5 6 7 8 9 ; do \
 			$(MKDIR) $${a}$${b}0 $${a}$${b}1 $${a}$${b}2 \
 				$${a}$${b}3 $${a}$${b}4 $${a}$${b}5 \
 				$${a}$${b}6 $${a}$${b}7 $${a}$${b}8 \
 				$${a}$${b}9 ; \
 			done ; done
-	$(CP) config.example $(LIBDIR)
-	$(CHOWN) -R news.news $(LIBDIR)/. $(SPOOLDIR)/.
-	$(CHMOD) 2750 $(SPOOLDIR)
+	$(CP) config.example $(DESTDIR)$(LIBDIR)
 	@echo edit /etc/inetd.conf to start $(BINDIR)/leafnode
 
 dist:
